etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/hapi/HapiMessageConverterHelperTest.groovy

```diff
@@ -0,0 +1,97 @@
+
+    Bundle mockBundle
+    Patient mockPatient
+
+    def setup() {
+        TestApplicationContext.reset()
+        TestApplicationContext.init()
+        TestApplicationContext.register(HapiMessageConverterHelper, HapiMessageConverterHelper.getInstance())
+        TestApplicationContext.injectRegisteredImplementations()
+
+        mockPatient = new Patient()
+        mockBundle = new Bundle().addEntry(new Bundle.BundleEntryComponent().setResource(mockPatient))
+    }
+
+    def "addEtorTag adds the ETOR message header tag to any Bundle"() {
+        given:
+        def expectedSystem = "http://localcodes.org/ETOR"
+        def expectedCode = "ETOR"
+        def expectedDisplay = "Processed by ETOR"
+
+        def messageHeader = new MessageHeader()
+        messageHeader.setId(UUID.randomUUID().toString())
+        def messageHeaderEntry = new Bundle.BundleEntryComponent().setResource(messageHeader)
+        mockBundle.getEntry().add(messageHeaderEntry)
+
+        when:
+        HapiMessageConverterHelper.getInstance().addEtorTagToBundle(mockBundle) as Bundle
+
+        then:
+        def messageHeaders = mockBundle.getEntry().get(1).getResource() as MessageHeader
+        def actualMessageTag = messageHeaders.getMeta().getTag()[0]
+
+        actualMessageTag.getSystem() == expectedSystem
+        actualMessageTag.getCode() == expectedCode
+        actualMessageTag.getDisplay() == expectedDisplay
+    }
+
+    def "addEtorTag adds the ETOR message header tag to any Bundle when message header is missing"() {
+        given:
+        def expectedSystem = "http://localcodes.org/ETOR"
+        def expectedCode = "ETOR"
+        def expectedDisplay = "Processed by ETOR"
+
+        when:
+        HapiMessageConverterHelper.getInstance().addEtorTagToBundle(mockBundle) as Bundle
+
+        then:
+        def messageHeaders = mockBundle.getEntry().get(1).getResource() as MessageHeader
+        def actualMessageTag = messageHeaders.getMeta().getTag()[0]
+
+        actualMessageTag.getSystem() == expectedSystem
+        actualMessageTag.getCode() == expectedCode
+        actualMessageTag.getDisplay() == expectedDisplay
+    }
+
+    def "addEtorTag adds the ETOR message header tag to any Bundle with existing Meta tag"() {
+        given:
+        def firstExpectedSystem = "a system"
+        def firstExpectedCode = "test"
+        def firstExpectedDisplay = "Processed by tests"
+        def secondExpectedSystem = "http://localcodes.org/ETOR"
+        def secondExpectedCode = "ETOR"
+        def secondExpectedDisplay = "Processed by ETOR"
+
+        def messageHeader = new MessageHeader()
+        messageHeader.setId(UUID.randomUUID().toString())
+        def meta = new Meta()
+        meta.addTag(new Coding(firstExpectedSystem, firstExpectedCode, firstExpectedDisplay))
+        messageHeader.setMeta(meta)
+        def messageHeaderEntry = new Bundle.BundleEntryComponent().setResource(messageHeader)
+        mockBundle.getEntry().add(messageHeaderEntry)
```

Comment:
Same question as above

Reply:
same.

Reply:
I added mockBundle = null as part of the setup(). This will ensure the mockBundle gets nulled out from the start, every time setup() runs.